storage/replica_raftstorage.go

```diff
@@ -90,8 +88,39 @@ func (r *Replica) Entries(lo, hi, maxBytes uint64) ([]raftpb.Entry, error) {
 	// stopping once we have enough.
 	var ents []raftpb.Entry
 	size := uint64(0)
+
+	hitEnts, hitSize, hitIndex := r.raftEntryCache.getEntry(lo, hi, maxBytes)
+	// Did the correct number of results come back? If so, we're all good.
+	if len(hitEnts) == int(hi)-int(lo) {
+		if TestingEntriesCacheHook != nil {
+			r.Lock()
+			TestingEntriesCacheHook(hitEnts)
```

Comment:
Instead of introducing this new hook, I would clear the cache between each test case and run each test twice: the first time misses the cache and goes to disk; the second hits the cache.